vulnerabilities/pagination_mixin.py

```diff
@@ -0,0 +1,58 @@
+class PaginatedListViewMixin:
+    paginate_by = 20
+    max_page_size = 100
+
+    PAGE_SIZE_CHOICES = [
+        {"value": 20, "label": "20 per page"},
+        {"value": 50, "label": "50 per page"},
+        {"value": 100, "label": "100 per page"},
+    ]
+
+    def get_paginate_by(self, queryset=None):
+        try:
+            page_size = int(self.request.GET.get("page_size", self.paginate_by))
+            if page_size <= 0:
+                return self.paginate_by
+            return min(page_size, self.max_page_size)
+        except (ValueError, TypeError):
+            return self.paginate_by
+
+    def get_context_data(self, **kwargs):
```

Comment:
It adds pagination data to template context - current page size, size choices (can be handled by altering PAGE_SIZE_CHOICES) , total count and page range. it also preserves search params across pagination.